substrate/frame/executive/src/lib.rs

```diff
@@ -356,16 +356,6 @@ where
 	pub fn try_runtime_upgrade(
 		checks: frame_try_runtime::UpgradeCheckSelect,
 	) -> Result<Weight, TryRuntimeError> {
-		if checks.try_state() {
```

Comment:
You should change the doc comment on this function if you remove this. It currently says you run try-state both before and after migrations.